src/develop/blend.c

```diff
@@ -225,8 +225,8 @@ static inline void _Lab_2_LCH(const float *Lab, float *LCH)
 static inline void _LCH_2_Lab(const float *LCH, float *Lab)
 {
   Lab[0] = LCH[0];
-  Lab[1] = cosf(2.0f * M_PI * LCH[2]) * LCH[1];
-  Lab[2] = sinf(2.0f * M_PI * LCH[2]) * LCH[1];
+  Lab[1] = cosf(2.0f * 3.141592654f * LCH[2]) * LCH[1];
```

Comment:
I suggest to use M_PI_F as a floating point constant at these places.

Reply:
Although, the macro `M_PI_F` is commonly used in the darktable OpenCL sources. It is not defined in any C source file. Note that not even `M_PI` is defined by any <i>standard</i> C header.

Possibly, one should add a new header file to the darktable sources, which defines new macros as `DT_M_PI`, `DT_M_PI_F` etc. for single and double precision representations of common mathematical constants.

Reply:
I replaced the numerical value 3.141592654f by the new macro `DT_M_PI_F`, which has been defined in the new header `common/math.h` and may be useful in other source code files as well.